index.js

```diff
@@ -30,6 +30,8 @@ const Socket = exports.Socket = class TCPSocket extends Duplex {
     this._pendingFinal = null
     this._pendingDestroy = null
 
+    this._timeout = null
```

Comment:
```suggestion
    this._timer = null
```

To avoid the `timout` and `_timeout` clash, which is a little confusing.